src/LeanCode.ContractsGenerator/Generation/TypeRefFactory.cs

```diff
@@ -62,6 +62,56 @@ public TypeRef From(ITypeSymbol symbol)
         }
     }
 
+    public NotificationTypeRef FromNotification(ITypeSymbol symbol)
+    {
+        return new()
+        {
+            Type = From(symbol),
+            Tag = GetFullTypeName(symbol),
+        };
+    }
+
+    private static string GetFullTypeName(ITypeSymbol typeSymbol)
```

Comment:
We can't rely on that, especially without conformance tests. Where does the algorithm come from? How can we be sure that it will generate the same thing as `Type.ToString`?